packages/guideline-blocks-settings/src/components/Attachments/types.ts

```diff
@@ -11,7 +17,8 @@ export type AttachmentsProps = {
     onUpload: (uploadedAttachments: Asset[]) => Promise<void>;
     onBrowse: (browserAttachments: Asset[]) => void;
     onSorted: (sortedAttachments: Asset[]) => void;
-};
+    triggerComponent?: (props: AttachmentsTriggerProps) => JSX.Element;
+} & ({ isOpen?: never; onOpenChange?: never } | { isOpen: boolean; onOpenChange: (isOpen: boolean) => void });
```

Comment:
I might be missing something but can this be 

```
{
    ...
    triggerComponent?: (props: AttachmentsTriggerProps) => JSX.Element;
    isOpen?: boolean;
    onOpenChange?: (isOpen: boolean) => void;
}
```

Reply:
I split the typing so that you can't have one without the other. Since whether or not the isOpen prop is defined determines whether the internal state is used at all it felt like a safer approach to guarantee that the component is either controlled or uncontrolled, but never in-between

Reply:
Cool, makes sense